### DISCUSSION TITLE
Tuning search parameters

### DISCUSSION THREAD

**Discussion by PeWu** [krisk/Fuse#574] [Q&A] [answered]
Hi,

I'm having trouble tuning the search parameters. I'm trying to replace the search function in [Topola Viewer](https://pewu.github.io/topola-viewer/) with fuse.js. The search is over people's names. I tried indexing a person's name as `'firstname lastname'` and `['firstname', 'lastname']` and both lead to unexpected results. Ideally, I wanted to be able to search for name prefixes, e.g. searching for `tom woz` would match `tomasz wozniak`. 

See this code sample for my examples: https://stackblitz.com/edit/typescript-y8hfkk?file=index.ts

In short, case 1 indexes:
```typescript
const list = [
  { name: ['tomasz', 'wozniak'] },
  { name: ['dominik', 'wozniak'] },
];
```
and the search for `tom wozniak` matches _dominik wozniak_ first.

Case 2 indexes:
```typescript
const list = [
  { name: 'tomasz wozniak' },
  { name: 'igor wozniak' }
];
```
and the search for `tom wozniak` matches _igor wozniak_ first.

Please help me tweak the indexing and search options. Or am I doing something wrong?

**Comment by krisk** [comment, marked as answer]
I would recommend you first transform your search list, separating the names into individual properties:

```js
const list = [
  { first: 'tomasz', lastName: 'wozniak' },
  { first: 'dominik', lastName: 'wozniak' }
];
```
